client/signup/steps/jpo-summary/index.jsx

```diff
@@ -169,19 +209,57 @@ class JPOSummaryStep extends React.Component {
 					</tbody>
 				</table>
 				<div>
-					<Button primary onClick={ this.completeOnboarding }>
+					<a
+						className="jpo-summary__visit-site button is-primary"
+						href={ get( this.props.signupDependencies, [ 'jpoConnect', 'queryObject', 'home_url' ] ) }>
 						{
-							translate( 'Visit your site' )
+							translate( 'Visit site' )
 						}
-					</Button>
+					</a>
 				</div>
 			</div>
 		);
 	}
 
+	componentWillMount() {
+		let choices = this.getOnboardingChoices();
+		const siteId = get( this.props.signupDependencies, [ 'jpoConnect', 'queryObject', 'client_id' ], -1 ),
+			hasBusinessInfo = some( choices.onboarding.businessInfo, i => '' !== i );
+		if ( 'store' === choices.onboarding.genre ) {
+			this.props.installPlugin( siteId, { slug: 'woocommerce', id: 'woocommerce/woocommerce' } );
+		}
+		if ( hasBusinessInfo ) {
+			choices = extend( {}, choices, {
+				widgets: true,
+			} );
+		}
+		this.props.updateSettings(
```

Comment:
maybe it comes in handy to move this logic into a new component's method with a clear name and have it called by `componentDidMount()` to keep this lifecycle method leaner.